src/renderer/constants.ts

```diff
@@ -29,19 +29,19 @@ export class ConstantProvider extends Blockly.zelos.ConstantProvider {
       } else {
         const style = {
           colourPrimary:
-            "colourQuaternary" in colour
+            "colourQuaternary" in colour && Boolean(colour.colourQuaternary)
```

Comment:
Without these checks, I'd get lots of `Invalid colour: "undefined"` errors. But is this code correct, @gonfunko ? As it just seems a little odd to be defining `Primary` `Secondary` and `Tertiary` always first trying to use the `Quaternary` setting..